common/djangoapps/enrollment/api.py

```diff
@@ -250,7 +251,30 @@ def get_course_enrollment_details(course_id):
         }
 
     """
-    return _data_api().get_course_enrollment_info(course_id)
+    cache_key = "enrollment.course.details.{course_id}".format(course_id=course_id)
+
+    try:
+        cached_enrollment_data = cache.get(cache_key)
+    except Exception:
+        # The cache backend could raise an exception (for example, memcache keys that contain spaces)
+        log.exception("Error occurred while retrieving course enrollment details from the cache")
+        cached_enrollment_data = None
+
+    if cached_enrollment_data:
+        log.info("Get enrollment data for course %s (cached)", course_id)
+        return cached_enrollment_data
+
+    course_enrollment_details = _data_api().get_course_enrollment_info(course_id)
+
+    try:
+        cache.set(cache_key, course_enrollment_details)
```

Comment:
What is the expiration timeout on this cache key?  I'd imagine we want to keep the expiration fairly short in case someone changes course enrollment dates in Studio.  Particularly in a moment of panic when the dates were entered incorrectly :)